tests/unit/backend/recipe/algorithm/test_PixelDiffractionCalibration.py

```diff
@@ -202,6 +204,19 @@ def test_reexecution_and_convergence(self):
             allOffsets.append(data["medianOffset"])
             assert allOffsets[-1] <= max(1.0e-4, allOffsets[-2])
 
+    def test_reference_pixel_selection(self):
+        """Test that the selected reference pixel is always a member of the detector group."""
+        algo = ThisAlgo()
+
+        # Test even and odd order groups;
+        #   test consecutive and non-consecutive groups.
+        # Do not test empty groups.
+        N_detectors = 10
+        dids = list(range(N_detectors))
```

Comment:
Can you add an extra test that includes a gap in `dids`?  For instance 
``` python
dids = [1, 2, 4, 5]
```
and make sure it doesn't return `3`?

Reply:
Sorry, and one more with e.g. `dids = [1, 2, 11]` and make sure it returns `2`?